java/common/common-task/src/main/java/sleeper/task/common/RunCompactionTasks.java

```diff
@@ -181,12 +182,13 @@ private static void launchTasks(
     /**
      * Create the container networking configuration.
      *
-     * @param  subnets the subnet name
-     * @return         task network configuration
+     * @param  instanceProperties the instance properties
+     * @return                    task network configuration
      */
-    private static NetworkConfiguration networkConfig(List<String> subnets) {
+    private static NetworkConfiguration networkConfig(InstanceProperties instanceProperties) {
         AwsVpcConfiguration vpcConfiguration = new AwsVpcConfiguration()
-                .withSubnets(subnets);
+                .withSubnets(instanceProperties.getList(SUBNETS))
+                .withSecurityGroups(instanceProperties.getList(ECS_SECURITY_GROUPS));
```

Comment:
It looks like this is only used for Fargate?

It's quite confusing because the launch type is checked twice, once in launchTasks and once in createRunTaskRequest. The network configuration is created in launchTasks regardless of the launch type, but it's used in createRunTaskRequest, and only if the launch type is Fargate.

Do we need a separate issue for EC2?

Reply:
Thanks, i've made it use the network configuration regardless of what launch type its using

Reply:
I don't think that'll work. In CompactionStack it defines the EC2 task with bridge network mode. If you check the documentation for the builder method to set the network configuration, it says it's just for awsvpc network mode.

I think we need to handle this for EC2 compaction in the CDK. This other pull request sets a security group for that explicitly:
- https://github.com/gchq/sleeper/pull/2839